### PR TITLE
2091 refactor program type

### PR DESCRIPTION
**Pull request by davidgisbey**
### Context

API V2 currently exposes program type directly to the frontend. This PR refactors this to expose a funding_type attribute.

Later PR will remap funding type to an attribute on the course model.

### Changes proposed in this pull request

- Expose a funding_type attribute which remaps to program type when it hits the backend.
- Program type is no longer exposed 

### Guidance to review

- Now has been de-prioritised. 

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

File: app/models/course.rb [hunk 1]
```diff
@@ -394,6 +394,27 @@ def is_published?
     content_status == :published
   end
 
+  def assign_program_type(funding_type)
+    case funding_type
+    when 'salary'
+      if !self_accredited?
+        update(program_type: :school_direct_salaried_training_programme)
+      else
+        errors.add(:program_type, "Fee is not valid for a self accredited course")
```
[the remark this answers is not in the document]
Comment (defong) [reply app/models/course.rb]: Maybe `Self accredited course can not be a salaried`
or `Salary is not valid for a self accredited course`

File: spec/models/course/assign_program_type_spec.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
+RSpec.describe Course, type: :model do
+  describe '#assign_program_type' do
+    before do
+      subject.assign_program_type(funding_type)
+    end
+
+    describe 'when funding type is salary' do
+      context 'an externally accredited courses' do
+        let(:subject) { create(:course, :with_accrediting_provider) }
+        let(:funding_type) { 'salary' }
```
[the remark this answers is not in the document]
Comment (davidgisbey) [reply spec/models/course/assign_program_type_spec.rb]: done

File: app/models/course.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (davidgisbey) [reply app/models/course.rb]: And done.

**Comment by faissalMT** [comment]
[Refactor program_type in edit options to hide business logic](https://trello.com/c/HMEGUlpt/2091-refactor-programtype-in-edit-options-to-hide-business-logic)

**Comment by faissalMT** [comment]
[Refactor program_type in edit options to hide business logic](https://trello.com/c/HMEGUlpt/2091-refactor-programtype-in-edit-options-to-hide-business-logic)

**Pull request merged**